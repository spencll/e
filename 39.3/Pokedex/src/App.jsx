import "./App.css";
import React from "react";
import Pokedex from "./Pokedex.jsx";

function App() {
  return (
    <>
      <Pokedex />
    </>
  );
}
export default App;
