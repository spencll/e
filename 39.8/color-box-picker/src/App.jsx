import "./App.css";
import BoxList from "./BoxList";

// ## **Testing Requirements**

// Make sure both of these apps have tests. Here are the guidelines for testing:
// - **Every component needs a smoke + snapshot test.**
// - **Critical Business Logic needs tests too**. Use your judgement for this, but make sure you cover core functionality. Examples include: the ability to enter data, submit a form, and have the desired outcome appear in the DOM.

// ## **Part 1 - Color Box Maker**

// Create a new React application, which contains the following components:

// ***App***
// this component should render the ***BoxList*** component.

// - When each ***Box*** component is displayed, add a button with the text of “X” next to each ***Box***. When this button is clicked, remove that specific box. This will require you to pass a function down as props - the button **should not** be a separate component, it should be included in the Box component.
function App() {
  return (
    <>
      <BoxList />
    </>
  );
}

export default App;
