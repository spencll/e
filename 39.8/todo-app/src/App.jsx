import "./App.css";
import TodoList from "./TodoList";

// ## **Part 2 - Todo App - List, Add, Remove Todos**
// Create a Todo App that allows users to see, add, edit, and remove todos. It should contain the following components.

function App() {
  // ***App***
  // this component should render the ***TodoList*** component

  return (
    <>
      <TodoList />
    </>
  );
}

export default App;
