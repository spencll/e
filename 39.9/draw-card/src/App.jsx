import { useState } from "react";
import "./App.css";
import Deck from "./Deck";

function App() {
  const [count, setCount] = useState(0);

  return (
    <>
      <Deck />
    </>
  );
}

export default App;
