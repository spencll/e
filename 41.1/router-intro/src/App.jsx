import './App.css'
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import Soda from './Soda'
import Chips from './Chips'
import Cookies from './Cookies'
import Home from './Home'

function App() {

  return (
    <>

    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home/>} />
          <Route path="/soda" element={<Soda/>} />
          <Route path="/chips" element={<Chips/>} />
          <Route path="/cookies" element={<Cookies/>} />
        </Routes>
      </BrowserRouter>
    </div>
    </>
  )
}

export default App
