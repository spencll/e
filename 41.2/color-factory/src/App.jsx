import {BrowserRouter, Routes, Route, Link, Navigate} from "react-router-dom"
import { useState } from "react";
import './App.css'
import ColorDetails from "./ColorDetails";
import ColorForm from "./ColorForm";
import ColorList from "./ColorList";

// ### **User Stories**

// 1. As a user, I can go to */colors* to see a list of all available colors.

// 2. As a user, I can click on one of the colors in my colors list and get taken to a page where I can see that color in all its glory.
//     (The route here should be */colors/:color* )
    
// 3. As a user, I can click on a button to show a form that will let me add a new color.
//     Note that you can give an *input* a type of *color* if you’re trying to select a color. (The route here should be */colors/new*)
    
// 4. As a user, when I submit my new color form, I am redirected to the colors index, and my new color appears at the top.
// 5. As a user, if I try to navigate to a color page that does not exist (eg, */colors/nope*), I am redirected to the colors index page.
// 6. As a user, if I try to navigate to an invalid url (eg, */this-is-not-valid*), I am redirected to the colors index page.

// Here’s an idea of what your app could look like:


function App() {

  // creating new state of colors
  const [colors, setColors] = useState(["red","yellow","green"]);

  // Add color function to pass down to ColorForm
  const addColor = (newColor) => {
    setColors([newColor, ...colors ]);
  };
  
  //  Form display for add color
  const [isFormDisplayed, setIsFormDisplayed] = useState(true);
    const toggleFormDisplay = () => {
      setIsFormDisplayed(!isFormDisplayed);
    };

  return (
    <>  
    <h1>COLORS!</h1>
   
    <BrowserRouter>
    {isFormDisplayed && <Link to="/colors/new" onClick={toggleFormDisplay}><h2>Add color</h2></Link>}
    <Routes>
    <Route exact path="/colors" element={<ColorList colors={colors} toggleFormDisplay={toggleFormDisplay}/>} ></Route>
    <Route exact path="/colors/new" element={<ColorForm addColor={addColor} toggleFormDisplay={toggleFormDisplay}/>}></Route>
    <Route path="/colors/:color" element={<ColorDetails colors={colors} toggleFormDisplay={toggleFormDisplay}/>}></Route>
    <Route path="*" element={<Navigate to="/colors" />} />

 </Routes>
</BrowserRouter>
    </>
  )
}

export default App
