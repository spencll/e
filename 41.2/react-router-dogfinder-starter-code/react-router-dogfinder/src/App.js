import './App.css';
import DogList from './DogList';
import DogDetails from './DogDetails';
import { useState, useEffect } from 'react';
import { Route, Routes, BrowserRouter, Navigate } from "react-router-dom";
import axios from 'axios';

// Outline
// App-> DogList -> DogDetails
// 1. Set up routes and links to the components (elements)
// 2. Set up elements 



// ## **Part 1: React Router Dog Finder**
// Build an app that routes to different dogs and displays information on that dog when you’re at that route.

// The routes should look like this:
// - */dogs* is the homepage and shows all three dogs
// - Clicking on a dog from the homepage takes you to that dog’s route. For example, clicking on Whiskey will take you to */dogs/whiskey*.
// - every other endpoint not listed should redirect you to */dogs.*


// ### **Recommended Structure**
// You can preload the *<App />* component with the following *defaultProps* for convenience:


// The *<App />* should render:
// - a *<Nav />* component with the dogs’ names passed as props
// - a *<Switch>* with your *<Route />* declarations


// Here is an Example Snippet from the `render` method of *<App />* To Get You Started:

// ```jsx
// return (
//   <Switch>
//     <Route exact path="/dogs" >
//       <DogList /> // what props will this need?
//     </Route>
//     <Route path="/dogs/:name" >
//       <DogDetails /> // what props will this need?
//     </Route>
//     <Redirect to="/dogs" />
//   </Switch>
// );
// ```

// In this example:

// - *<DogList />* takes all the dog info from the props of *<App />*
// - *<DogDetails />* shows all of the info about a single dog
// - In *<DogDetails />*, how will you derive the current dog, e.g. *whiskey*?
// - Bonus: is there a way to get the current dog *before* you render the component, passing *dog* instead of the entire list of dog data?



function App() {

  // creating new state of dogs
    const [dogs, setDogs] = useState({
      data: null,
      isLoading: true
    });

// Fetching dogs from db.json upon first render 
useEffect(() => {
  async function loadDogs(){
    const response = await axios.get("http://localhost:5001/dogs")
    setDogs({
      data: response.data,
      isLoading: false
    })
  }
  loadDogs()
}, [])

// dogs is object with array of dogs in data and true/false in isLoading

  return (
    <>
    <BrowserRouter>
    <Routes>
        {/* what props will DogList? dogs */}
    <Route exact path="/dogs" element={<DogList dogs={dogs.data}/>} ></Route>
        {/* what props will DogDetails need? dogs, can pull name,age,src,facts */}
    <Route path="/dogs/:name" element={<DogDetails dogs={dogs.data}/>}></Route>
    {/* No matching route, default here */}
    <Route path="*" element={<Navigate to="/dogs" />} />

 </Routes>
</BrowserRouter>
    </>
  );


}


export default App;
