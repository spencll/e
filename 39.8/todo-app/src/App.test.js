import React from "react";
import { render } from "@testing-library/react";
import App from "./App.jsx";


test("App renders", () => {
  render(<App />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});