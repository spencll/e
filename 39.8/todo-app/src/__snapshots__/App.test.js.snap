// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todos
    </h3>
    <form>
      <label
        for="todo"
      >
        Add todo:
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="todo"
        type="text"
        value=""
      />
      <button>
        Add todo
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
